lib/matplotlib/axes/_base.py

```diff
@@ -1287,25 +1287,59 @@ def apply_aspect(self, position=None):
 
     def axis(self, *v, **kwargs):
         """
-        Convenience method for manipulating the x and y view limits
-        and the aspect ratio of the plot. For details, see
-        :func:`~matplotlib.pyplot.axis`.
+        Call signature::
+
```

Comment:
So, please correct me if I am wrong, but it looks like there should never be more than one arg as the logic is

```python
if len(v) == 1 and isinstance(str, v[0]):
     # parse string
elif not len(v):
    # look work for kwargs
else:
   v = v[0]
   assert len(v) == 4
   # do stuff with the vector
```

so I think a docstring like
```python
If no positional or kwargs are given, returns the current axes limits are returned with no side effects.

If a positional argument is passed the keyword arguments are ignored.

If no positional argument is passed, the keyword arguments are used.

Parameters
----------
v : list of float or {'on', 'off', 'equal', 'tight', 'scaled', 'normal', 'auto', 'image', 'square'}
    Optional positional argument

    Value to set the axis settings to.  If a list it is interpreted as [xmin, xmax, ymin, ymax].
    If a string 

       INSERT TABLE HERE

emit : bool, optional
    Passed to  set_{x,y}lim functions, if observers are notified of axis limit change

xmin, ymin, xmax, ymax : float, optional
    The axis limits to be set
```